packages/jest-mock/src/index.js

```diff
@@ -691,6 +691,65 @@ class ModuleMockerClass {
     return object[methodName];
   }
 
+  spyOnProperty(object: any, propertyName: any, accessType = 'get'): any {
+    if (typeof object !== 'object' && typeof object !== 'function') {
```

Comment:
this should more closely match the normal `spyOn`.

Reply:
actually this `if` is the same of the `spyOn` above. what do you mean? i have adapted the code to with with a property descriptor instead of a normal property. do i have missed something here?

Reply:
Stuff like "// IE 8 doesn't support `definePropery` on non-DOM nodes" etc is not needed.

But anyways, can we make `jest.spyOn` work, so this new function is unnecessary?

Reply:
keep in mind that making `spyOn` work with getters/setters will introduce a breaking change between `jasmine` and `jasmine-light` shipped with `jest`, adding a little extra step during migrations to `jest`.
for this reason, i actually prefer following the original project and create a new method `spyOnProperty` for that.